alphadia/workflow/optimization.py

```diff
@@ -95,24 +96,42 @@ def step(
                 verbosity="progress",
             )
             return
+        if record_step:
+            try:
+                # Takes the target number of precursors if at least the target is found,
+                precursor_cutoff_idx = (
+                    precursors_df.sort_values(by="precursor_idx")
+                    .iloc[
+                        2
```

Comment:
sure you want to pick only one index? or is there a `:` missing ;-)

Reply:
In this case, we just want the correct cutoff index, so only one index is wanted.

Reply:
all good, maybe introduce `cutoff_index = 2 * .. ` then and add a little explanation (e.g. on the factor of 2 ;-))